File: platform/android-customer/app/src/main/java/ca/shawcliffe/tomsproduce/Email.kt
package ca.shawcliffe.tomsproduce

/** Mirrors platform/web/src/lib/email.ts so all surfaces validate identically. */
object Email {
    private val regex = Regex("^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$")

    fun error(raw: String): String? {
        if (raw.isEmpty()) return null
        return if (!regex.matches(raw)) "Enter a valid email address (e.g. jane@example.com)" else null
    }
}

File: clients/little-dough-bread-co/ios-customer/ShawcliffeCustomer/Email.swift
import Foundation

/// Mirrors platform/web/src/lib/email.ts so both surfaces validate identically.
enum Email {
    private static let regex = try! NSRegularExpression(pattern: "^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$")

    static func error(for raw: String) -> String? {
        guard !raw.isEmpty else { return nil }
        let range = NSRange(raw.startIndex..., in: raw)
        return regex.firstMatch(in: raw, range: range) == nil ? "Enter a valid email address (e.g. jane@example.com)" : nil
    }
}
